### PR TITLE
POC: use styled aria-label instead of titles

### PR DESCRIPTION
**Pull request by roperzh**
OK to merge.

Short review.

Summary of changes:

@taylorpoe suggested in our last meeting the idea of using tool-tips instead of the titles so we can style and control the delay until they appear.

This is a random idea to avoid using a `<Tooltip />` component, mainly because I have the feeling that will be so cumbersome to add a title in that way that nobody will do it.

So, the idea is to style elements with the attribute `aria-label` and add a tooltip with pseudo elements (this is not an idea of mine at all).

Where this approach falls short:

- A tooltip may have z-index problems, because we are tied to the stacking context of the target element.
- The position of the tooltip is not dynamically updated

If we decide that this is a good idea I can clean up and test this properly.

PS: @taylorpoe agreed to buy a new toy to Jiggs if we get to implement tooltips, so let's push this forward! (_note: he didn't_)

Completed checkin tasks:

- [x] Did manual testing of interrelated functionality
- [ ] Wrote an automated test for existing and modified functionality
- [ ] Added measurement instrumentation (Mixpanel, etc.)

### CODE REVIEW

File: packages/haiku-timeline/package.json [hunk 1]
```diff
@@ -22,6 +22,7 @@
   },
   "dependencies": {
     "@haiku/core": "3.5.2",
+    "balloon-css": "^0.5.0",
```
Comment (stristr) [inline packages/haiku-timeline/package.json:25]: Unused dep?

File: packages/haiku-ui-common/src/react/TooltipStyles.tsx [hunk 2]
```diff
@@ -0,0 +1,206 @@
+import * as React from 'react';
```
Comment (stristr) [inline packages/haiku-ui-common/src/react/TooltipStyles.tsx:1]: If you want to just put this in a stylesheet, e.g. in `haiku-ui-common/assets/common.css`, you could probably just put this in (e.g.) `haiku-glass/index.html` like:

`<link rel="stylesheet" href="../../node_modules/haiku-ui-common/assets/common.css" />`

Should just work.

**Review by stristr — COMMENTED** [review]
Nice! One note added with an alternative to using React to host the stylesheet.

**Comment by taylorpoe** [comment]
Sweet!

File: packages/haiku-ui-common/src/react/TooltipStyles.tsx [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+        /* Base styles for the entire tooltip */
+        [aria-label]:before,
+        [aria-label]:after {
+          position: absolute;
+          visibility: hidden;
+          opacity: 0;
+          transition: opacity 0.2s ease-in-out,
+            visibility 0.2s ease-in-out,
+            transform 0.2s cubic-bezier(0.71, 1.7, 0.77, 1.24);
```
Comment (taylorpoe) [inline packages/haiku-ui-common/src/react/TooltipStyles.tsx:25]: This cu-bez's overshoot feels a bit off to me (in the gif at least). Could we knock this back to just an `ease`?

File: packages/haiku-timeline/package.json (hunk 1 above)
Comment (roperzh) [reply packages/haiku-timeline/package.json:25]: yes, apologies for that!

File: packages/haiku-ui-common/src/react/TooltipStyles.tsx (hunk 2 above)
Comment (roperzh) [reply packages/haiku-ui-common/src/react/TooltipStyles.tsx:1]: This is gold, thank you!

File: packages/haiku-ui-common/src/react/TooltipStyles.tsx (hunk 3 above)
Comment (roperzh) [reply packages/haiku-ui-common/src/react/TooltipStyles.tsx:25]: that makes sense, will do! thanks!

**Comment by tsortwehttam** [comment]
Closing in favor of #718 , which now contains these commits.

**Pull request closed without merging**